packages/flutter/test/material/outline_button_test.dart

```diff
@@ -1043,6 +1043,97 @@ void main() {
     _checkPhysicalLayer(buttonElement, fillColor.withAlpha(0x00));
   });
 
+  testWidgets('OutlineButton respects the provided materialTapTargetSize', (WidgetTester tester) async {
+    final Key key1 = UniqueKey();
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: Material(
+          child: Center(
+            child: OutlineButton(
+              key: key1,
+              materialTapTargetSize: MaterialTapTargetSize.padded,
+              child: const SizedBox(width: 50.0, height: 8.0),
+              onPressed: () { },
+            ),
+          ),
+        ),
+      ),
+    );
+
+    expect(tester.getSize(find.byKey(key1)), const Size(88.0, 48.0));
+
+    final Key key2 = UniqueKey();
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: Material(
+          child: Center(
+            child: OutlineButton(
+              materialTapTargetSize: MaterialTapTargetSize.shrinkWrap,
+              key: key2,
+              child: const SizedBox(width: 50.0, height: 8.0),
+              onPressed: () { },
+            ),
+          ),
+        ),
+      ),
+    );
+
+    expect(tester.getSize(find.byKey(key2)), const Size(88.0, 36.0));
+
+    final Key key3 = UniqueKey();
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: Material(
+          child: Center(
+            child: OutlineButton.icon(
+              key: key3,
+              materialTapTargetSize: MaterialTapTargetSize.padded,
+              icon: const Icon(Icons.add_alarm),
+              label: const SizedBox(width: 50.0, height: 8.0),
+              onPressed: () { },
+            ),
+          ),
+        ),
+      ),
+    );
+
+    final Size addAlarmIconSize = tester.getSize(find.byIcon(Icons.add_alarm));
+
+    // The expected size width consist out of the width of SizedBox (50),
```

Comment:
This kind of comment is really important, glad to see that you've taken a crack at it.  It would help to include similar comments, anywhere that the sizes aren't obvious.
```dart
// The expected width is the sum of:
//   the width of the icon
//   the gap between the icon and the label (8)
//   the width of the label (50)
//   the horizontal padding: start 12, end 16
```